pkg/mount/block/blockdev.go

```diff
@@ -498,6 +498,32 @@ func (b BlockDevices) FilterPartType(guid string) BlockDevices {
 	return b.FilterNames(names...)
 }
 
+// FilterPartLabel returns a list of BlockDev objects whose underlying block
+// device has the given parition label.
+func (b BlockDevices) FilterPartLabel(label string) BlockDevices {
+	var names []string
+	for _, device := range b {
+		table, err := device.GPTTable()
+		if err != nil {
+			continue
+		}
+		for i, part := range table.Partitions {
+			if part.IsEmpty() {
+				continue
+			}
+			if strings.ToLower(part.Name()) == strings.ToLower(label) {
+				r := []rune(device.Name[len(device.Name)-1:])
+				if unicode.IsDigit(r[0]) {
+					names = append(names, fmt.Sprintf("%sp%d", device.Name, i+1))
```

Comment:
I'm a bit confused by what is going on here. Are you assuming a device ending in a number (ex: nvme0n1) have partitions numbered like nvme0n1p1, nvme0n1p2, ... and devices ending in a letter (ex: sda) have partitions numbered like sda1, sda2, ...? Perhaps add a comment.

Reply:
Yes, I guess so. Took that part from the existing functions FilterPartID and FilterPartType.

Reply:
@rminnich Do you know if this is always consistent with how Linux numbers partitions?

Reply:
ping @rminnich